src/fuselage/CCPACSCrossBeamAssemblyPosition.cpp

```diff
@@ -24,9 +46,132 @@ CCPACSCrossBeamAssemblyPosition::CCPACSCrossBeamAssemblyPosition(CCPACSCargoCros
 {
 }
 
+std::string CCPACSCrossBeamAssemblyPosition::GetDefaultedUID() const
+{
+    return GetUID();
+}
+
+PNamedShape CCPACSCrossBeamAssemblyPosition::GetLoft()
+{
+    return PNamedShape( new CNamedShape(GetGeometry(true), GetDefaultedUID()) );
+}
+
+TiglGeometricComponentType CCPACSCrossBeamAssemblyPosition::GetComponentType() const
+{
+    return TIGL_COMPONENT_GENERICSYSTEM | TIGL_COMPONENT_PHYSICAL;
+}
+
 void CCPACSCrossBeamAssemblyPosition::Invalidate()
 {
-    invalidated = true;
+    for (size_t i=0; i<2; i++) {
+        m_geometry[i] = boost::none;
+    }
+    m_cutGeometry = boost::none;
+}
+
+TopoDS_Shape CCPACSCrossBeamAssemblyPosition::GetGeometry(bool just1DElements, TiglCoordinateSystem cs)
+{
+    if (!m_geometry[just1DElements]) {
+        BuildGeometry(just1DElements);
+    }
+
+    TopoDS_Shape shape = m_geometry[just1DElements].value();
+    if (cs == GLOBAL_COORDINATE_SYSTEM) {
+        CTiglTransformation trafo = m_parent->GetParent()->GetParent()->GetTransformationMatrix();
+        return trafo.Transform(shape);
+    }
+    else {
+        return shape;
+    }
+}
+
+TopoDS_Shape CCPACSCrossBeamAssemblyPosition::GetCutGeometry(TiglCoordinateSystem cs)
+{
+    if (!m_cutGeometry) {
+        BuildCutGeometry();
+    }
+
+    TopoDS_Shape shape = m_cutGeometry.value();
+    if (cs == GLOBAL_COORDINATE_SYSTEM) {
+        CTiglTransformation trafo = m_parent->GetParent()->GetParent()->GetTransformationMatrix();
+        return trafo.Transform(shape);
+    }
+    else {
+        return shape;
+    }
+}
+
+void CCPACSCrossBeamAssemblyPosition::BuildGeometry(bool just1DElements)
+{
+    if (just1DElements) {
+        // the crossBeam is an edge resulting from the intersection between the face of a frame and a plane
+        CCPACSFrame& frame = m_uidMgr->ResolveObject<CCPACSFrame>(m_frameUID);
+
+        const TopoDS_Shape frameGeometry = frame.GetGeometry(true, FUSELAGE_COORDINATE_SYSTEM);
```

Comment:
@bernhardmgruber When opening your example file `fuselage_structure.xml` with Tigl Viewer and displaying the cross beams via` Draw -> Aircraft -> Draw any Component` I get the above mentioned error. I think it fails when trying to get the frame geometry. Could you please take a look into it? Thanks!

Reply:
will do

Reply:
i believe this is fixed with the new frame geometry generation implemented as part of #402. Waiting for this PR to be accepted.